compiler/src/dmd/nogc.d

```diff
@@ -86,6 +92,11 @@ public:
         }
         else if (fd.ident == Id._d_arrayappendT || fd.ident == Id._d_arrayappendcTX)
         {
+            if (checkOnly)
+            {
+                err = true;
+                return;
```

Comment:
With a nested function, you can remove the duplicated branches which aren't covered by tests.